odbc_catalog.c

```diff
@@ -3834,9 +3834,33 @@ retrieve_table_from_db_class (int cci_connection, char *table_name, T_CCI_ERROR
   int cci_request;
 
   char *sql_statment = "SELECT class_name FROM db_class WHERE class_name = ?";
-  char *param_list[] = { table_name };
+  char tablename [512];
+  char *param_list[] = { tablename };
+  char query_buf [512];
+  char *query = sql_statment;
+  char *p;
+  char owner [256];
+
+  p = strchr (table_name, '.');
+  if (p)
+    {
+      strcpy (owner, table_name);
+      strcpy (tablename, p + 1);
```

Comment:
It seems to me that the code is unnecessary.

Reply:
Thank you for your comment, I'll rewrite it soon.

Reply:
* original table_name is the form public.code
  * line #3848 will copy '**code**' into _tablename_.
  * and line #3863 will make owner from 'public.code' to 'public'

What's your opinion?

Reply:
I didn't see using tablename in param_list . my mistake.

Reply:
got it!